### ISSUE TITLE
etcdserver/security:  impossible to disable security once it's been turned on

### ISSUE THREAD

**Issue by mgoelzer** [coreos/etcd#2527]
Once I enable security in etcd, I cannot disable it.  Error message is "Can't delete root user."

You can reproduce the problem like this:

````
$ curl -L http://127.0.0.1:4001/v2/security/enable -XPUT -d '{"user":"root","password":"pass"}'
$ curl --user root:pass -L http://127.0.0.1:4001/v2/security/enable -XGET
{"enabled":true}
$ curl --user root:pass -L http://127.0.0.1:4001/v2/security/enable -XDELETE
{"message":"Can't delete root user; disable security instead."}
````

From looking through commit 001efa0639657f5aa4e7f6e627f9ed814da4f714, it appears to me that it's actually impossible to disable security once it's been turned on.  The reason is that DisableSecurity() calls DeleteUser("root"), but the first thing DeleteUser() does is explicitly prevent you from deleting root.  :smiley:

**Comment by mgoelzer** [comment]
I modified DeleteUser() to allow removal of root (just comment out the first 3 lines), and now everything works as expected.  You can enable and disable security just as described in the docs.

Shall I make a PR for this?

**Comment by yichengq** [comment]
Based on user feedback, our current plan is to decouple enable/disable and user(root) setting. @barakmich will guide the road to that.

**Comment by mgoelzer** [comment]
Ah, ok, that makes sense @yichengq

**Comment by barakmich** [comment]
Fixed in #2654
